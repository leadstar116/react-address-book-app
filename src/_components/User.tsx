import React, { useState } from 'react'
import { UserInfo } from '../_constants/users.interface'
import { Row, Col } from 'react-bootstrap'
import DetailsModal from './DetailsModal'

type Props = {
    data: UserInfo
}

const User = (props: Props) => {
    const [showDetail, setShowDetail] = useState(false);

    const handleClick = () => {
        setShowDetail(true)
    }
    return (
        <>
            <Row className="m-1 rounded border align-items-center text-center p-1 user" onClick={handleClick}>
                <Col xs={12} sm={2}><img src={props.data.picture.thumbnail} alt=""/></Col>
                <Col xs={12} sm={2}><span>{ props.data.name.username }</span></Col>
                <Col xs={12} sm={2}><span>{ props.data.nat }</span></Col>
                <Col xs={12} sm={2}><span>{ props.data.name.first + ' ' + props.data.name.last }</span></Col>
                <Col xs={12} sm={4}><span>{ props.data.email }</span></Col>
            </Row>
            <DetailsModal
                show={showDetail}
                onHide={() => setShowDetail(false)}
                user={props.data}
            />
        </>
    )
}

export default User