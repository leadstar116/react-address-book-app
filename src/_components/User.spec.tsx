import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { UserInfo } from '../_constants/users.interface';
import User from './User'


describe('User Component', () => {
    const testUserInfo = {
        email: 'test@test.com',
        name: {
            first: "James",
            last: "Bond",
            username: "test"
        },
        picture: {
            large: "https://randomuser.me/api/portraits/men/75.jpg",
            medium: "https://randomuser.me/api/portraits/med/men/75.jpg",
            thumbnail: "https://randomuser.me/api/portraits/thumb/men/75.jpg"
        } ,
        cell: "+14325432870",
        dob: {
            date: "1993-07-20T09:44:18.674Z",
            age: 26
        },
        gender: "male",
        location: {
            street: {
                name: "9278 new road",
                number: 12
            },
            city: "kilcoole",
            state: "waterford",
            postcode: 93027,
            country: "CH",
            coordinates: {
              latitude: "20.9267",
              longitude: "-7.9310"
            },
            timezone: {
              offset: "-3:30",
              description: "Newfoundland"
            }
        },
        nat: "CH",
        phone: "+14325432870",
        id: {
            name: "testuser111",
            value: "12"
        },
        login: {
            username: "test"
        }
    } as UserInfo

    let component = renderer.create(
        <BrowserRouter>
            <User data={testUserInfo}/>
        </BrowserRouter>
    );

    it('should render with given props', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
})