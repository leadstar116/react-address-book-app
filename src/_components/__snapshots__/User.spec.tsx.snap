// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Component should render with given props 1`] = `
<div
  className="m-1 rounded border align-items-center text-center p-1 user row"
  onClick={[Function]}
>
  <div
    className="col-sm-2 col-12"
  >
    <img
      alt=""
      src="https://randomuser.me/api/portraits/thumb/men/75.jpg"
    />
  </div>
  <div
    className="col-sm-2 col-12"
  >
    <span>
      test
    </span>
  </div>
  <div
    className="col-sm-2 col-12"
  >
    <span>
      CH
    </span>
  </div>
  <div
    className="col-sm-2 col-12"
  >
    <span>
      James Bond
    </span>
  </div>
  <div
    className="col-sm-4 col-12"
  >
    <span>
      test@test.com
    </span>
  </div>
</div>
`;
